### PR TITLE
Setting up boilerplate

### PR DESCRIPTION
**Pull request by trungh13**
- Fix the file structures
- Add modules : axios, prop-types, eslint
- Add mock-up reducers/authReducer, actions/types

### CODE REVIEW

**Comment by pandaren89** [comment]
Code is fine without eslint stuff. But lots of warnings when eslint is installed in root level.

How do you suggest we proceed ?

File: package.json [hunk 1]
```diff
@@ -21,6 +23,7 @@
     "eject": "react-scripts eject"
   },
   "devDependencies": {
+    "eslint": "^4.19.1",
```
Comment (Nizar-Rahme) [inline package.json]: let's reconfigure eslint so we don't have to add it and remove every time we deploy

File: src/App.js [hunk 2]
```diff
@@ -1,19 +1,35 @@
...
+  return <div>{routes}</div>;
 };
 
-export default App;
+App.propTypes = {
+  auth: PropTypes.shape({}).isRequired,
+};
+
+const mapStateToProps = state => ({
+  auth: state.auth,
```
[the remark this answers is not in the document]
Comment (trungh13) [reply src/App.js]: solved

File: package.json (hunk 1 above)
Comment (trungh13) [reply package.json]: now I still keep the eslint but no more warning when build / compile

**Pull request merged**